core/services/relay/evm/chain_reader.go

```diff
@@ -205,6 +244,39 @@ func (cr *chainReader) addDecoderDef(contractName, methodName string, outputs ab
 		return err
 	}
 	output.mod = mod
-	cr.parsed.decoderDefs[wrapItemType(contractName, methodName, false)] = output
+	cr.parsed.decoderDefs[WrapItemType(contractName, methodName, false)] = output
 	return output.Init()
 }
+
+func setupEventInput(event abi.Event, def types.ChainReaderDefinition) ([]abi.Argument, *codecEntry, map[string]bool) {
+	topicFieldDefs := map[string]bool{}
+	for _, value := range def.EventInputFields {
+		capFirstValue := abi.ToCamelCase(value)
+		topicFieldDefs[capFirstValue] = true
+	}
+
+	filterArgs := make([]abi.Argument, 0, maxTopicFields)
+	info := &codecEntry{}
+	indexArgNames := map[string]bool{}
+
+	for _, input := range event.Inputs {
+		if !input.Indexed {
+			continue
+		}
+
+		filterWith := topicFieldDefs[abi.ToCamelCase(input.Name)]
```

Comment:
Is it necessary to camel case it? Do we have to allow naming disparity between config and ABI? If not we can just do `slices.Contains(def.EventInputFields, input.Name)` here so we don't have to init topicFieldDefs

Reply:
It's what the `abi` library does, so it's expecting you to do it to.  I think we should be consistent with that library.  Also, it makes it a lot easier to use, since the naming convention isn't the same on solidity as go, it would make people require a lot more `rename` modifications in config otherwise.